master/internal/kubernetes/pod.go

```diff
@@ -101,9 +153,191 @@ func (p *pod) startPod(ctx *actor.Context) error {
 	}
 }
 
-func (p *pod) configureResourcesRequirements() v1.ResourceRequirements {
-	return v1.ResourceRequirements{
-		Limits: map[v1.ResourceName]resource.Quantity{
+func (p *pod) receivePodStatusUpdate(ctx *actor.Context, msg podStatusUpdate) error {
+	p.pod = msg.updatedPod
+
+	switch msg.updatedPod.Status.Phase {
+	case k8sV1.PodPending:
+		// When pods are deleted, Kubernetes sometimes transitions pod statuses
+		// to pending prior to deleting them. We ignore this state if we have
+		// triggered the deletion.
+		if p.resourcesDeleted {
+			ctx.Log().WithField("pod", p.podName).Info(
+				"ignoring pod status because resources are being deleted")
+			return nil
+		}
+
+		containerState := getContainerState(msg.updatedPod.Status.Conditions)
+		if containerState == container.Running {
+			ctx.Log().WithField("pod", p.podName).Errorf(
+				"unexpected containers status while pod is pending")
+		}
+
+		if containerState != p.container.State {
+			if containerState == container.Starting {
+				// Kubernetes does not have an explicit state for pulling container
+				// images. We insert it here because our  current implementation of
+				// the trial actor requires it.
+				ctx.Log().WithField("pod", p.podName).Infof(
+					"transitioning pod state from %s to %s",
+					p.container.State, container.Pulling)
+				p.container = p.container.Transition(container.Pulling)
+
+				rsc := sproto.ContainerStateChanged{Container: p.container}
+				ctx.Tell(p.taskHandler, rsc)
+			}
+
+			ctx.Log().WithField("pod", p.podName).Infof(
+				"transitioning pod state from %s to %s",
+				p.container.State, containerState)
+			p.container = p.container.Transition(containerState)
+
+			rsc := sproto.ContainerStateChanged{Container: p.container}
+			ctx.Tell(p.taskHandler, rsc)
+		}
+
+	case k8sV1.PodRunning:
+		if p.container.State != container.Running {
+			p.container = p.container.Transition(container.Running)
+
+			logStreamer, err := newPodLogStreamer(p.podInterface, p.podName, ctx.Self())
+			if err != nil {
+				return err
+			}
+			var ok bool
+			p.logStreamer, ok = ctx.ActorOf(fmt.Sprintf("%s-logs", p.podName), logStreamer)
```

Comment:
non-blocking: If you get rid of `logStreamer` (see other comments), then this can just become `_, ok :=`.

Reply:
done.